C/c4Observer.cc

```diff
@@ -32,14 +32,21 @@ namespace litecore {
         ,_callback(move(callback))
         {
             _collection->sequenceTracker().useLocked<>([&](SequenceTracker &st) {
-                _notifier.emplace(st,
+                _notifier.emplace(&st,
                                   [this](CollectionChangeNotifier&) {_callback(this);},
                                   since);
             });
         }
 
 
         ~C4CollectionObserverImpl() {
+            if (!_collection->isValid()) {
+                // HACK: If the collection is not valid anymore, the notifier tracker is probably
```

Comment:
Why is this? tracker is owned by C4CollectionImpl object. If the collection object is good (in c++ sense, namely it's deleted yet), tracker should also be good, or not? _collection->isValid() == false means it's relation with the underlying database is broken. Did you see its crashing after you retained the collection object?

Reply:
Yes, it will crash when destroying the notifier unless I do this.

Reply:
This is what I explained in the 2nd point in the PR comment.

Reply:
Oh, that's because you retained only in C4CollectionObserverImpl. The notifier doesn't hold the ref count. Notifier needs to retain it as well.

Reply:
It has nothing to do with the retain count of the collection.  When the collection is invalidated, the tracker is destroyed.

Reply:
Ah, I see, close() is called if it's not valid. We destruct all members before the collection object itself is deleted. In other words, only thing from retained collection object is query if it's valid. Hopefully, this, the tracker, is the only sub-object that is leaked out from the collection obj.